web/modules/custom/joinup_group/joinup_group.module

```diff
@@ -262,6 +263,54 @@ function joinup_group_rdf_entity_update(EntityInterface $entity) {
   }
 }
 
+/**
+ * Implements hook_ENTITY_TYPE_update().
+ *
+ * Make sure that when an entity is (un)pinned it is reindexed and its caches
+ * are invalidated.
+ */
+function joinup_group_meta_entity_update(MetaEntityInterface $entity): void {
+  if ($entity->bundle() === 'pinned_in') {
+    $parent_entity = $entity->getTargetEntity();
+    if (empty($parent_entity)) {
+      // If the parent entity no longer exists nothing needs to be done.
+      return;
+    }
```

Comment:
You say now that this should not happen, but it is not the responsibility of this hook implementation to verify the integrity of the relationships between meta entities and their parent entities.

What this code is responsible for is to index the parent entity if the meta entity changes. If this doesn't exist (because the return type definition of `MetaEntityInterface::getTargetEntity()` tells us that it can possibly return `NULL`), then we do not have enough information to decide whether this is wrong or is expected behavior.

Any function that we write should fully check its inputs and handle its data responsibly. It should not introduce undefined behavior, like knowingly indexing a `NULL` value, randomly clearing the entire entity static cache by passing an empty value to `::resetCache()`, or cause a fatal error by trying to call `::getCacheTagsToInvalidate()` on a `NULL` object.

Reply:
Yes, this is the key phrase:

> (...) but it is not the responsibility of this hook implementation to verify the integrity of the relationships between meta entities and their parent entities.

Then why are we checking? IMHO, we should not. This piece of code should not be in business of checking this consistency. And yes, we should should get a fatal error because something very bad happened.